src/org/labkey/workflow/WorkflowController.java

```diff
@@ -1,1238 +1,1239 @@
+        }
+
+        public void setIncludeCompletedTasks(boolean includeCompletedTasks)
+        {
+            _includeCompletedTasks = includeCompletedTasks;
+        }
+    }
+
+    /**
+     * Retrieves the process diagram associated with a particular processInstance or processName as a png.
+     * If there is no such diagram, a plain text response will be returned indicating that there is no such image.
+     */
+    @RequiresPermission(ReadPermission.class)
+    public class ProcessDiagramAction extends BaseViewAction
+    {
+        @Override
+        protected String getCommandClassMethodName()
+        {
+            return null;
+        }
+
+        @Override
+        public ModelAndView handleRequest() throws Exception
+        {
+            InputStream stream = null;
+            String contentType = "image/png";
+            if (getViewContext().getRequest().getParameter("processInstanceId") != null)
+                stream = WorkflowManager.get().getProcessDiagram(getViewContext().getRequest().getParameter("processInstanceId"));
+            else if (getViewContext().getRequest().getParameter("processDefinitionKey") != null)
+                stream = WorkflowManager.get().getProcessDiagramByKey(getViewContext().getRequest().getParameter("processDefinitionKey"), getContainer());
+            if (stream == null)
+            {
+                contentType = "text/plain";
+                stream = new ByteArrayInputStream("Unable to retrieve process diagram.  Perhaps you need to deploy the process.".getBytes("UTF-8"));
+            }
+            byte[] imageBytes = IOUtils.toByteArray(stream);
+            HttpServletResponse response = getViewContext().getResponse();
+
+            response.setContentType(contentType);
+            response.setContentLength(imageBytes.length);
+            response.getOutputStream().write(imageBytes);
+            return null;
+        }
+
+        @Override
+        public void validate(Object o, Errors errors)
+        {
+            HttpServletRequest request = getViewContext().getRequest();
+            if (request.getParameter("processInstanceId") == null && request.getParameter("processDefinitionKey") == null)
+                errors.reject(ERROR_MSG, "Either a process instance id or a process definition key must be provided");
+        }
+    }
+
+
+    /**
+     * Shows the data about a process instance if the user has permissions to see this task
+     */
+    @RequiresPermission(ReadPermission.class)
+    public class ProcessInstanceDataAction extends ReadOnlyApiAction<ProcessInstanceDetailsForm>
+    {
+        private WorkflowProcess _processInstance;
+
+
+        @Override
+        public void validateForm(ProcessInstanceDetailsForm form, Errors errors)
+        {
+            if (form.getProcessInstanceId() == null)
+                errors.rejectValue("processInstanceId", ERROR_MSG, PROCESS_INSTANCE_ID_MISSING);
+            else
+            {
+         
```

Comment:
@labkey-susanh 
Should RemoveProcessInstanceAction extend MutatingApiAction, and have UpdatePermission? since it is deleting a process, changing server state?